### ISSUE TITLE
Single instance vs one instance per db call. What's the proper way to use this library?

### ISSUE THREAD

**Issue by AZ1904** [praeclarum/sqlite-net#1001]
Currently i am using sqlite-net-sqlcipher with ASP.NET core and this is how i register the library (notice the `AddSingleton`):

`services.AddSingleton<SQLiteAsyncConnection>(_ =>
                new SQLiteAsyncConnection(new SQLiteConnectionString(dbPath, true, key: "ABCD1234")));`

And use it in the following manner:

```
internal class AbcdRepository : IAbcdRepository
{
    private readonly SQLiteAsyncConnection _database;

    public AbcdRepository(SQLiteAsyncConnection database)
    {
        _database = database;
    }

    public async Task<IReadOnlyCollection<IAbcd>> ReadAll()
    {
        return await _database.Table<Abcd>().ToListAsync().ConfigureAwait(false);
    }
}
```

That is, I keep one single connection and leave it open through out the lifetime of the application. Is this the how the library is intended to be used? Or should i create and dispose a new connection every time i need to make a database call? like the following:

```
internal class AbcdRepository : IAbcdRepository
{
    public async Task<IReadOnlyCollection<IAbcd>> ReadAll()
    {
       using(var database = new SQLiteAsyncConnection(...)){
        return await database.Table<Abcd>().ToListAsync().ConfigureAwait(false);
       }
    }
}
```

**Comment by sjlombardo** [comment]
@AZ1904 - in general with SQLCipher you will want to use fewer connections for performance reasons. It would not be appropriate to create a new connection every time you make a database call. In that case, the overhead of performing key derivation for every connection will slow things down considerably.

**Comment by ske66** [comment]
I agree, repository patterns usually work best with a singleton class. Use Dependency Injection and limit your connections, keep the connection alive
